src/deadline/vred_submitter/default_vred_job_template.yaml

```
@@ -587,6 +592,17 @@ steps:
                 os.environ[CODE_PASSING_ENV_VAR] = VRED_PYTHON_BOOTSTRAP_CODE
                 os.environ[FLEXLM_DIAGNOSTICS_ENV_VAR] = FLEXLM_DIAGNOSTICS_HIGH_VALUE
 
+            def reader_thread(pipe, queue):
+                """
+                Thread to read output from pipe and put it into queue
+                """
+                while True:
+                    line = pipe.readline()
+                    if not line:
+                        break
+                    queue.put(line)
+                pipe.close()
```

We might want to add some sort of handling if the Queue is full and returns the `Full` exception. If possible, we might want to read the thread exit code/error code to make sure that the thread terminated gracefully and not because of an exception

Any other sort of IPC would work as well to allow for the reader thread to indicate any potential exceptions to the main thread

---

Good point, I have added a `error_queue` that allows the reader thread to communicate exceptions, and queue overflow warnings to the main thread.

1. Queue Full Handling: Added `except queue.Full` blocks with error reporting
2. Thread Error Communication:  
   1. Added a separate `error_queue` for IPC to report thread exceptions to the main thread
   2. The main thread now monitors both thread aliveness and error queue for exception reporting
3. Graceful Degradation: When error queue is full, the system falls back to console output rather than failing silently